Replicator/tests/ReplicatorSGTest.cc

```diff
@@ -85,6 +85,34 @@ class ReplicatorSGTest : public ReplicatorAPITest {
 
 TEST_CASE_METHOD(ReplicatorSGTest, "API Auth Failure", "[.SyncServer]") {
     _sg.remoteDBName = kProtectedDBName;
+    
+    SECTION("No Credentials") { }
+    
+    SECTION("Wrong Credentials") {
+        Encoder enc;
+        enc.beginDict();
+            enc.writeKey(C4STR(kC4ReplicatorOptionAuthentication));
+            enc.beginDict();
+                enc.writeKey(C4STR(kC4ReplicatorAuthType));
+                enc.writeString("Basic"_sl);
+                enc.writeKey(C4STR(kC4ReplicatorAuthUserName));
+                enc.writeString("brown");
+                enc.writeKey(C4STR(kC4ReplicatorAuthPassword));
+                enc.writeString("sugar");
+                enc.writeKey(C4STR(kC4ReplicatorAuthEnableChallengeAuth));
+        
+            SECTION("Preemptive Auth") {
+                enc.writeBool(false);
```

Comment:
Will this trigger Assert?

Reply:
What Assert?

Reply:
If it's the same assert as @borrrden mentioned, the answer is YES.